ARCs/arc-0023.md

```diff
@@ -0,0 +1,76 @@
+---
+arc: 23
+title: Sharing Application's information
+description: Append informations to the compiled application's bytes
+author: Stéphane Barroso (@sudoweezy)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/80
+status: Draft
+type: Meta
+created: 2023-01-11
+---
+
+## Abstract
+
+The following document introduces a convention for appending information to the compiled application's bytes. 
+The goal of this convention is to standardize the process of verifying and interacting with smart contracts. The encoded byte is `arc23` followed by the length of the CID representing a folder of files & the CID itself.
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+> Comments like this are non-normative.
+
+### CID
+The <a href="https://github.com/multiformats/cid">CID</a> allows to access the corresponding files using <a href="https://docs.ipfs.tech/">IPFS</a>.
+
+The CID **MUST** represent one file or a folder of files.
+The minimum requirement for this ARC is a file representing the ABI (ABI as described in [ARC-4](arc-0004.md))
+
+### Encoding
+The encoded byte is `arc23` followed by the length of the CID, the number of files, and the CID itself.
+
+eg: for the example stored in `/asset/arc-0023/Folder`, 
+`arc23 59 2 bafybeihdzhfifv46p27ee6weqq6h7ydttzwni7gvdogr3mg6rwscdyzici`
+`bafybeihdzhfifv46p27ee6weqq6h7ydttzwni7gvdogr3mg6rwscdyzici` correspond to this Folder: `../assets/arc-0023/Folder`. It can be obtained with `ipfs add --cid-version=1 Folder/*  -q -w -n| tail -1`
+
+If there is only one file, the CID **MUST** be related to the ABI of the application. 
+eg: `arc23 59 1 bafkreiajguyi3i5hf4h7blslsv53rjbexzzvpkyhdf26hmuqthrk4iko4a`
+`bafkreiajguyi3i5hf4h7blslsv53rjbexzzvpkyhdf26hmuqthrk4iko4a` correspond to this Abi File: `../assets/arc-0023/Folder/Abi.json`. It can be obtained with `ipfs add --cid-version=1 Folder/Abi.json -q`
+
+> The option -n is for obtaining the hash without pinning the file, you **MAY** remove the option in production
+
+Otherwise, if there is more than one file, the CID must correspond to a folder of files. The ABI **MUST** be accessible by querying and IPFS node with `{cid}/Abi.json`
+eg:`bafybeihdzhfifv46p27ee6weqq6h7ydttzwni7gvdogr3mg6rwscdyzici/Abi.json`.
+
+### Appending
```

Comment:
```suggestion
### Including the encoded information to the program
```

Reply:
Also one question is whether the appending should be done at the end of the clear program and/or approval program.
Since clear programs are less space constrained in general, maybe we should put them there? What do you think?